### PR TITLE
Remove prerelease logic for `next` site

### LINKED ISSUE #1261: Remove "next" site
### Contributions

- [x] I have read the contribution docs, and understand what's expected of me.

### What page(s) or areas on docs.getdbt.com are affected?

Now that docs are versioned, we no longer need the twin deployments at docs.getdbt.com and next.docs.getdbt.com... right?

### What changes are you suggesting?

We can remove logic put by yours truly in:
- `docusaurus.config.js`
- `netlify.toml` (`PRERELEASE = "true"`)
- Netlify to deploy the `next` branch
- Algolia, which runs a separate search index for the "next" site... that never worked _quite_ right

Let's also:
- update the PR template, to mention `VersionBlock` instead of the `next` branch
- set up a CNAME redirect from next.docs.getdbt.com to docs.getdbt.com
- just have one branch named `main` :)

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by jtcohen6**
Resolves #1261

## Description & motivation
Undoes everything from https://github.com/dbt-labs/docs.getdbt.com/pull/306:
- Remove `netlify.toml`
- Remove `PRERELEASE`-related logic from `docusaurus.config.js`
- Update PR template to recommend new approach to docs versioning

### TODOs
Out of band for this PR:
- Update CNAME redirect for next.docs.getdbt.com — should point to docs.getdbt.com, rather than alt Netlify deploy
- Rename default branch in this repo to `main`, to match our standards everywhere else

### CODE REVIEW

**Review by JKarlavige — APPROVED** [review]
lgtm! For the TODOs, is there anything I can do to help with those?

**Comment by jtcohen6** [comment]
Thanks @JKarlavige!
- I think you know better than I do how to update CNAMEs these days. This used to be an "ask Connor" :)
- Let's just double-check that @runleonarun is cool with switching the default from `current` to `main`. That can also happen separately. I realize there are still two spots in `docusaurus.config.js` where we rely on the branch name being `current`.

**Comment by JKarlavige** [comment]
> I think you know better than I do how to update CNAMEs these days. This used to be an "ask Connor" :)

The process is still very close to that, but now it's asking the Infra team 😆  I will take care of that. Are we clear to setup that redirect now, or should I wait for a green light from you?

**Comment by jtcohen6** [comment]
Let's set it up now! The `next` site hasn't been any different from "current" docs.getdbt.com since Dec 6

**Comment by JKarlavige** [comment]
Sounds good! Jira ticket opened here: https://dbtlabs.atlassian.net/browse/IST-138

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -12,14 +12,10 @@ Include any notes about things that need to happen before this PR is merged, e.g
 - [ ] Ensure PR #56 is merged
 -->
 
-## Pre-release docs
-Is this change related to an unreleased version of dbt?
```
Comment (runleonarun) [inline .github/pull_request_template.md:16]: Love this change! ❤️

**Review by runleonarun — APPROVED** [review]
:shipit:

**Comment by jtcohen6** [comment]
Going to wait on merging this, just to be safe, until @JKarlavige has had a chance to sync with Infra about deleting/switching the DNS records for next.docs.getdbt.com

**Comment by JKarlavige** [comment]
> Going to wait on merging this, just to be safe, until @JKarlavige has had a chance to sync with Infra about deleting/switching the DNS records for next.docs.getdbt.com

Hey @jtcohen6, next is now redirecting to docs.getdbt.com. This PR is safe to merge.

**Pull request merged**